In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
data = pd.read_excel("C:/Users/RAHUL REDDY/Desktop/AIML/Letsupgrade/AI-ML-July-2020-master/Day-12/data12.xlsx")
data

DistanceFromHome_Yes  DistanceFromHome_No  MonthlyIncome_Yes  \
0                      0                    6             131160   
1                     10                    0              41890   
2                     17                    0             193280   
3                      0                    2              83210   
4                     10                    0              23420   
5                      0                    8              40710   
6                     11                    0              58130   
7                     18                    0              31430   
8                      0                    1              20440   
9                      0                    7             134640   
10                    17                    0              79910   
11                    28                    0              33770   
12                    14                    0              55380   
13                     0                    1              57620   
14                     0                    1              25920   
15                     0                    1              53460   
16                     0                    3              42130   
17                     0                    1              41270   
18                     0                    7              24380   
19                     0                    8              68700   

    MonthlyIncome_No  TotalWorkingYears_Yes  TotalWorkingYears_No  \
0              20000                      1                     2   
1             120000                      6                     8   
2              55000                      5                     0   
3              25000                     13                     4   
4              89000                      9                     7   
5             140000                     28                    12   
6              21000                      5                    14   
7              78000                     10                     3   
8              89000                     10                     1   
9              25000                      6                     4   
10            124000                     21                     8   
11             78000                     16                    12   
12             21000                     37                     7   
13             45000                     10                     6   
14            110000                      5                    10   
15             78000                      7                     3   
16            120000                      3                     9   
17             48000                     15                    11   
18             12000                     10                     7   
19             42000                      8                     6   

    YearsAtCompany_Yes  YearsAtCompany_No  YearsWithCurrManager_Yes  \
0                    1                  2                         0   
1                    5                  4                         4   
2                    5                  1                         3   
3                    8                  3                         5   
4                    6                  4                         4   
5                    7                  5                         7   
6                    0                  1                         0   
7                    0                  2                         0   
8                    9                  0                         8   
9                    6                  5                         5   
10                  20                  4                        10   
11                  15                  5                        11   
12                  36                 12                        13   
13                  10                  1                         9   
14                   5                 15                   

In [4]:
from scipy.stats import mannwhitneyu

# Attrition Vs Distance from Home

In [5]:
data.columns

Index(['DistanceFromHome_Yes', 'DistanceFromHome_No', 'MonthlyIncome_Yes',
       'MonthlyIncome_No', 'TotalWorkingYears_Yes', 'TotalWorkingYears_No',
       'YearsAtCompany_Yes', 'YearsAtCompany_No', 'YearsWithCurrManager_Yes',
       'YearsWithCurrM0anager_No'],
      dtype='object')

In [6]:
h0 = data.DistanceFromHome_Yes
h1 = data.DistanceFromHome_No
stats,p = mannwhitneyu(h0,h1)
print(stats,p)

192.0 0.4140521132569824


__As the P value of 0.414 is > 0.05, the H0 is accepted and H1 is rejected.  
H0: There is significant differences in the Distance From Home between attrition (Y) and attirition (N).  
Ha: There is no significant differences in the Distance From Home between attrition (Y) and attirition (N).__

# Attrition Vs Income

In [7]:
h0 = data.MonthlyIncome_Yes
h1 = data.MonthlyIncome_No
stats,p = mannwhitneyu(h0,h1)
print(stats,p)

186.0 0.35743877353947784


__As the P value of 0.357 is > 0.05, the H0 is accepted and H1 is rejected.  
H0: There is significant differences in the MonthlyIncome between attrition (Y) and attirition (N).  
Ha: There is no significant differences in the MonthlyIncome between attrition (Y) and attirition (N).__

# Attrition Vs Total Working Years

In [8]:
h0 = data.TotalWorkingYears_Yes
h1 = data.TotalWorkingYears_No
stats,p = mannwhitneyu(h0,h1)
print(stats,p)

138.5 0.04902245434345275


__As the P value of 0.049 is > 0.05, the H0 is accepted and H1 is rejected.  
H0: There is significant differences in the TotalWorkingYears between attrition (Y) and attirition (N).  
Ha: There is no significant differences in the TotalWorkingYears between attrition (Y) and attirition (N).__

# Attrition Vs Years at company

In [10]:
h0 = data.YearsAtCompany_Yes
h1 = data.YearsAtCompany_No
stats,p = mannwhitneyu(h0,h1)
print(stats,p)

135.5 0.04074481303429345


__As the P value of 0.040 is < 0.05, the H0 is rejected and H1 is accepted.  
H0: There is no significant differences in the YearsAtCompany between attrition (Y) and attirition (N).  
Ha: There is significant differences in the YearsAtCompany between attrition (Y) and attirition (N).__

# Attrition Vs YearsWithCurrentManager

In [13]:
h0 = data.YearsWithCurrManager_Yes
h1 = data.YearsWithCurrM0anager_No
stats,p = mannwhitneyu(h0,h1)
print(stats,p)

178.5 0.2838510217877289


__As the P value of 0.283 is > 0.05, the H0 is accepted and H1 is rejected.  
H0: There is significant differences in the YearsWithCurrManager between attrition (Y) and attirition (N).  
Ha: There is no significant differences in the YearsWithCurrManager between attrition (Y) and attirition (N).__

# Statistical Tests (Separate T Test)

# Attrition Vs Distance From Home

In [15]:
from scipy.stats import ttest_ind
data.columns

Index(['DistanceFromHome_Yes', 'DistanceFromHome_No', 'MonthlyIncome_Yes',
       'MonthlyIncome_No', 'TotalWorkingYears_Yes', 'TotalWorkingYears_No',
       'YearsAtCompany_Yes', 'YearsAtCompany_No', 'YearsWithCurrManager_Yes',
       'YearsWithCurrM0anager_No'],
      dtype='object')

In [17]:
z1=data.DistanceFromHome_Yes
z2=data.DistanceFromHome_No
stat, p=ttest_ind(z2,z1)
print(stat, p)

-1.9274328359269242 0.061425596197977474


__As the P value is again 0.06, which is > than 0.05, the H0 is accepted and ha is rejected.  
H0: There is significant differences in the Distance From Home between attrition (Y) and attirition
(N).  
Ha: There is no significant differences in the Distance From Home between attrition (Y) and attirition (N)__

# Attrition Vs Income

In [24]:
z1=data.MonthlyIncome_Yes
z2=data.MonthlyIncome_No
stat, p=ttest_ind(z2, z1)
print(stat, p)

0.36633494816189316 0.7161469849563733


__As the P value is again 0.71, which is > than 0.05, the H0 is accepted and ha is rejected.  
H0: There is  significant differences in the Monthly Income between attrition (Y) and attirition (N).  
Ha: There is no significant differences in the Monthly Income between attrition (Y) and attirition (N)__

# Attrition Vs Yeats At Company

In [20]:
z1=data.YearsAtCompany_Yes
z2=data.YearsAtCompany_No
stat, p=ttest_ind(z2, z1)
print(stat, p)

-1.5098569118878218 0.13935107509148328


__As the P value is again 0.13, which is > than 0.05, the H0 is acceptedand ha is rejecpted.  
H0: There is  significant differences in the Years At Company between attrition (Y) and attirition
(N).  
Ha: There is no significant differences in the Years At Company between attrition (Y) and attirition (N)__

# Attrition Vs Years With Current Manager

In [22]:
z1=data.YearsWithCurrManager_Yes
z2=data.YearsWithCurrM0anager_No
stat, p=ttest_ind(z2, z1)
print(stat, p)

0.4701903336937378 0.6409065062011281


__As the P value is again 0.64, which is > than 0.05, the H0 is accepted and ha is rejected.  
H0: There is significant differences in the Years With Current Manager between attrition (Y) and
attirition (N).  
Ha: There is no significant differences in the Years With Current Manager between attrition (Y) and
attirition (N)__